### PR TITLE
Implement GetVertexAttrib

### PR DESCRIPTION
**Pull request by dzbarsky**
@emilio

---
This change is [Reviewable](https://reviewable.io/reviews/servo/gleam/79)

### CODE REVIEW

**Comment by dzbarsky** [comment]
needs changes...

File: src/gl.rs [hunk 1]
```diff
@@ -1112,6 +1112,24 @@ pub fn get_program_iv(program: GLuint, pname: GLenum) -> GLint {
...
+        let mut result: GLint = 0 as GLint;
+        ffi::GetVertexAttribiv(index, pname, &mut result);
+        return result;
+    }
+}
+
+#[inline]
+pub fn get_vertex_attrib_fv(index: GLuint, pname: GLenum) -> Vec<GLfloat> {
+    unsafe {
+        let mut result = vec![0 as GLfloat; 4];
```
Comment (dzbarsky) [inline src/gl.rs]: something about this function isn't right. I'm getting this crash:
```
 │ thread 'WebGLThread' panicked at 'Unexpected WebGL error: 0x502 (1282)', /Users/dzbarsky/mozilla/webrender_traits/src/webgl.rs:442
  │ stack backtrace:
  │    1:        0x10a94f4f8 - std::sys::backtrace::tracing::imp::write::h714760a4c8c0cdd8
  │ FAIL gl.getVertexAttrib(0, gl.CURRENT_VERTEX_ATTRIB)[0] should be 1. Was 0.
  │    2:        0x10a9557d5 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hff309ab1d83ffd90
  │    3:        0x10a9553ee - std::panicking::default_hook::h08ad3bb09872855b
  │    4:        0x10a0dfc2e - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::he7fa94d82e246627
  │    5:        0x10a93d0b8 - std::sys_common::unwind::begin_unwind_inner::h406d5f1a330b854b
  │    6:        0x10a93e9ee - std::sys_common::unwind::begin_unwind_fmt::h57ea3fbee1a40196
  │    7:        0x1095b2791 - webrender_traits::webgl::WebGLCommand::apply::hf26127a7e3bda5c2
  │    8:        0x1095ae7e2 - canvas::webgl_paint_thread::WebGLPaintThread::handle_webgl_message::h27afef3930098b0d
```

from this: 
```
gl.vertexAttrib1fv(ii, [1]);
shouldBe('gl.getVertexAttrib(' + ii + ', gl.CURRENT_VERTEX_ATTRIB)[0]', '1');
``

If I take out the `ffi:GetVertexAttribfv` call and return all 0's, no crash

Comment (dzbarsky) [inline src/gl.rs]: I don't see anything wrong with the way I call ffi and cargo points to http://doc.servo.org/src/gleam/home/servo/buildbot/slave/doc/build/target/debug/build/gleam-e640bc53dabe9ccf/out/gl_bindings.rs.html#4328-4345 which doesn't seem very helpful. Any ideas @emilio ?

Comment (dzbarsky) [inline src/gl.rs]: 1282 is INVALID_OPERATION

Comment (dzbarsky) [inline src/gl.rs]: opengl says "If the value of SHADER_COMPILER is not TRUE, then the error INVALID_OPERATION is generated."

Comment (dzbarsky) [inline src/gl.rs]: although that might be for GetShaderPrecisionFormat

Comment (emilio) [inline src/gl.rs]: The calls seem correct, but we seem to need to raise an error on `getVertexAttrib(0, CURRENT_VERTEX_ATTRIB)`, from the OpenGL man page:

```
Generic vertex attribute 0 is unique in that it has no current state, so an error will be generated if index is 0.
```

Comment (emilio) [inline src/gl.rs]: Actually, that only happens in some opengl versions, not gles2, so we'll have to emulate that functionality I gues. I looked at https://www.opengl.org/sdk/docs/man/docbook4/xhtml/glGetVertexAttrib.xml for the avobe statement.

Comment (emilio) [inline src/gl.rs]: In the [OpenGL 2.1 spec](https://www.opengl.org/sdk/docs/man2/xhtml/glGetVertexAttrib.xml) (the mode we open the OpenGL context for normal servo in desktop IIRC):

```
GL_INVALID_OPERATION is generated if index is 0 and pname is GL_CURRENT_VERTEX_ATTRIB.
```

So... This changes are correct, we need to emulate that somehow in the WebGL layer, not here.

**Comment by dzbarsky** [comment]
Nice sleuthing, that's a lot more reasonable than the explanation I found.  Looks like the OGL man pages are a lot easier to read than the kronos spec, thanks for pointing those out.

**Pull request merged**